dissect/database/sqlite3/sqlite3.py

```diff
@@ -74,9 +123,16 @@ def __init__(self, fh: BinaryIO, wal_fh: BinaryIO | None = None):
 
         self.page = lru_cache(256)(self.page)
 
-    def open_wal(self, fh: BinaryIO) -> None:
+    def open_wal(self, fh: Path | BinaryIO) -> None:
         self.wal = WAL(fh)
 
+    def checkpoints(self) -> Iterator[SQLite3]:
+        if not self.wal:
+            return
+
+        for checkpoint in self.wal.commits:
+            yield SQLite3(self.fh, self.wal.fh, checkpoint)
```

Comment:
```suggestion
            yield SQLite3(self.fh, self.wal, checkpoint)
```
Together with the other suggestion, then we don't have to re-parse the WAL.